private-templates-service/client/src/store/currentTemplate/actions.ts

```diff
@@ -101,15 +104,60 @@ function requestTemplateFailure(): CurrentTemplateAction {
   }
 }
 
-export function updateTemplate(templateID?: string, currentVersion?: string, templateJSON?: string, sampleDataJSON?: string, templateName?: string, state?: PostedTemplate.StateEnum, tags?: string[], isShareable?: boolean) {
+function failureUpdateCurrentTemplateVersion(): CurrentTemplateAction {
+  return {
+    type: FAILURE_UPDATE_CURRENT_TEMPLATE_VERSION,
+    text: "failure to update current template version"
+  }
+}
+
+function receiveUpdateCurrentTemplateVersion(version?: string, templateJSON?: object, sampleDataJSON?: object): CurrentTemplateAction {
+  return {
+    type: RECEIVE_UPDATE_CURRENT_TEMPLATE_VERSION,
+    text: "receive update current template version",
+    templateJSON: templateJSON,
+    sampleDataJSON: sampleDataJSON,
+    version: version,
+  }
+}
+
+function requestUpdateCurrentTemplateVersion(): CurrentTemplateAction {
+  return {
+    type: REQUEST_UPDATE_CURRENT_TEMPLATE_VERSION,
+    text: "request update current template version",
+  }
+}
+
+export function updateCurrentTemplateVersion(template: Template, version: string) {
+  return function (dispatch: any) {
+    dispatch(requestUpdateCurrentTemplateVersion());
+    if (template.instances) {
+      let numInstances = template.instances.length;
+      for (let j = 0; j < numInstances; j++) {
+        if (template.instances[j] && template.instances[j].version && typeof (template.instances[j].version) !== 'undefined' && template.instances[j].version == version) {
```

Comment:
Do we have to say `typeof (template.instances[j].version) !== 'undefined'`? Can't we just say `template.instances[j].version`?